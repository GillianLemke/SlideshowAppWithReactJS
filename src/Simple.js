import React from 'react';
//import { createStore } from 'redux';
//import ReactDOM from 'react-dom';
import './index.css';
import $ from 'jquery';
import {clickTitle} from './actions/slideactions';

//let store = createStore()
export default class Simple extends React.Component {

  getPageInfo() {
    return $.getJSON('google.com/slide')
      .then((data) => {
        // how can we get data for individual variables?
        this.setState({ title:data.results })
      });
  }

  makeList(contentList) {
    return(
      <ul>
        {contentList.map((item, index) => <li key={index}>{item}</li>)}
      </ul>
    );
    // for (var i = 0; i < contentList.length; i++) {
    //   return (<li>{contentList[i]}</li>);
    // }
    // return(<div></div>);
  }
  render() {
    return (
      <div>
        <div className="simple-slide">
          <h1 onClick={clickTitle}>{this.props.simpleData.title}</h1>

            {this.makeList(this.props.simpleData.content)}
        </div>

      </div>
    );
  }

}
