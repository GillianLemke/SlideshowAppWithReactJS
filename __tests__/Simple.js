import React from 'react'
import Simple from '../src/Simple.js'
import {shallow} from 'enzyme' // could also mount
import toJson from 'enzyme-to-json'

describe('Simple', () => {
  it('renders', () => {
    const wrapper = shallow(<Simple title='' type='' contents={[]} nextClick={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('responds to next click', () => {
    const nextListener = jest.fn()
    const wrapper = shallow(<Simple title='' type='' contents={[]} nextClick={nextListener} />)
    wrapper.find('.Next-Button').simulate('click') // also could do event ex. target, value
    expect(nextListener).toHaveBeenCalled()
  });
  it('responds to previous click', () => {
    const prevListener = jest.fn()
    const wrapper = shallow(<Simple title='' type='' contents={[]} prevClick={prevListener} />)
    wrapper.find('.Prev-Button').simulate('click') // also could do event ex. target, value
    expect(prevListener).toHaveBeenCalled()
  });
  it('checks title', () => {
    const wrapper = shallow(<Simple title='cool title' type='' contents={['hello', 'bonjour']} />)
    wrapper.find('h1')
    expect(toJson(wrapper))==='cool title'
  });
  it('list method renders', () => {
    const contentList = ['hello', 'bonjour']
    const wrapper = shallow(<Simple title='cool title' type='' contents={contentList} />)
    expect(toJson(wrapper))==='hello'
    expect(wrapper.find('ol').length) >= 1
  });

});
