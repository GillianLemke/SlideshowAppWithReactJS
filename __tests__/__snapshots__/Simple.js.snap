exports[`Simple renders 1`] = `
<div>
  <div
    className="simple-slide">
    <h1
      onClick={[Function]}>
      simple title
    </h1>
    <ul>
      <li>
        hello
      </li>
      <li>
        bonjour
      </li>
    </ul>
  </div>
</div>
`;
